sky/resources.py

```diff
@@ -1290,6 +1291,22 @@ def is_empty(self) -> bool:
     def copy(self, **override) -> 'Resources':
         """Returns a copy of the given Resources."""
         use_spot = self.use_spot if self._use_spot_specified else None
+
+        current_override_configs = self._cluster_config_overrides
+        if self._cluster_config_overrides is None:
+            current_override_configs = {}
+        new_override_configs = override.pop('_cluster_config_overrides', {})
+        overlaid_configs = skypilot_config.overlay_skypilot_config(
+            original_config=config_utils.Config(current_override_configs),
+            override_configs=new_override_configs,
+        )
+        override_configs = config_utils.Config()
+        for key in constants.OVERRIDEABLE_CONFIG_KEYS_IN_TASK:
+            elem = overlaid_configs.get_nested(key, None)
+            if elem:
```

Comment:
```suggestion
            if elem is not None:
```
We don't know type of `elem`, could be a non-None falsy value like 0 or False.

Reply:
ack, was able to reproduce a problematic scenario without adding this check in.